### PR TITLE
Turn on add_runtime_assertion by default

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #102671

### CODE REVIEW

**Review by angelayi — APPROVED** [review]
Looks good to me! I'm landing the constraint cleanup diff rn though, so you'll have to rebase https://github.com/pytorch/pytorch/pull/102666

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D46343555).

File: torch/_export/__init__.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def export(
     f: Callable,
     args: Tuple[Any],
     constraints: Optional[List[Constraint]] = None,
+    *,
+    _turn_off_export_validity=False,
```
Comment (avikchaudhuri) [inline torch/_export/__init__.py:96]: Rename to `_add_runtime_assertions=True`? Or were you imagining other uses?

Also, in general, prefer "disable" over "turn_off"

File: torch/_export/passes/add_runtime_assertions_for_constraints_pass.py [hunk 2]
```diff
@@ -112,28 +113,36 @@ def _insert_specialized_shapes_assert(self, arg, dims, name, current_inp):
                 f"specialized at {shapes[dim]}"
             )
             dim_node = super().call_operator(
-                torch.ops.aten.sym_size,
+                torch.ops.aten.sym_size.int,
                 (arg, dim),
                 {},
-                NodeMetadata({}),
+                NodeMetadata({
```
Comment (avikchaudhuri) [inline torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:119]: Looks like this is repeated everywhere, maybe make a util? You can probably use a different function if you want to cut off the util call from the stack trace.

**Review by avikchaudhuri — APPROVED** [review]
Pending nits in comments.

File: torch/_export/__init__.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/_export/__init__.py:96]: Sure sounds good

File: torch/_export/passes/add_runtime_assertions_for_constraints_pass.py (hunk 2 above)
Comment (tugsbayasgalan) [reply torch/_export/passes/add_runtime_assertions_for_constraints_pass.py:119]: Yep can do.

**Pull request closed without merging**